src/Middleware/RateLimiting/src/RateLimitingMiddleware.cs

```diff
@@ -149,11 +149,20 @@ private LeaseContext CombinedAcquire(HttpContext context)
                 return new LeaseContext() { GlobalRejected = false, Lease = endpointLease };
             }
         }
-        catch (Exception)
+        catch (Exception ex)
         {
             endpointLease?.Dispose();
             globalLease?.Dispose();
-            throw;
+            // Don't throw if the request was canceled - instead log. 
+            if (ex is OperationCanceledException && context.RequestAborted.IsCancellationRequested)
```

Comment:
Nit: Why not make this its own `catch (OperationCanceledException ex)` block and move the disposals to a `finally`? It feels like that would be a lot simpler to follow.

Reply:
Because we don't want to dispose on the success path